src/Shell/ResultInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Shell;
+
+/**
+ * Result for shell command.
+ */
+interface ResultInterface
```

Comment:
Because it's not a good practice to make the high coupling between our code and third-party class.
If something will change in the third-party library it can break our code in many places. 
But when we use our interface all changes can be fixed in one class.

Reply:
This library uses Semantic Versioning, we should rely on it

Reply:
For example:  if magento 2.4 will require Symfony Console 4+ how Semantic Versioning will help us? some methods in Process class will be deprecated and we can use them at that moment in our code.

Reply:
We have a bunch of 3rd-party libs in use, once we'll see breaking changes - we can think about introducing some adapter-compatiblity

Reply:
At some point, it can be very hard to add an adapter or even impossible. Better to prevent such situations, especially if we want to add some extensibility in our code.